crates/turbo-tasks-fs/src/lib.rs

```diff
@@ -474,17 +488,21 @@ impl FileSystem for DiskFileSystem {
     ) -> Result<CompletionVc> {
         let full_path = self.to_sys_path(fs_path).await?;
         let content = content.await?;
-        let old_content = fs_path
-            .read()
-            .await
-            .with_context(|| format!("reading old content of {}", full_path.display()))?;
 
-        if *content == *old_content {
+        // Track the file, so that we will rewrite it if it ever changes.
+        fs_path.track().await?;
+
+        // We perform an untracked read here, so that this write is not dependent on the
+        // read value (and the memory it holds). It's possible the read can be freed if
+        // no other task tries to read it, which is entirely likely for a output file.
```

Comment:
Why "possible"? This will always happen when `old_content` goes out of scope. It's impossible that anyone else will hold a reference to it since it will created just for us in `read_file`. We do not share the memory with anyone. Neither is rust a GC language.

Reply:
This was a leftover comment from my first attempt at this, where I used `path.read().into_untracked().await?` directly. Because it was a Vc's value, it could be held by others that also `.read()`, but that's not the case with the current impl. Removed.